Data-Structures/LinkedList/src/main/java/LinkedList/NodeConstructor.java

```diff
@@ -0,0 +1,15 @@
+package LinkedList;
+
+import javax.annotation.Nullable;
+
+public class NodeConstructor<generic> {
+
+    public NodeConstructor reference;
+    public generic value;
+
+    public NodeConstructor(generic value, @Nullable NodeConstructor reference) {
```

Comment:
Objects start with Capital letters. Instead of 'generic', 'Generic' would be better.